src/Http/Url.php

```diff
@@ -68,9 +68,8 @@ public function match($patterns)
             $patterns = [$patterns];
         }
 
-        //Remove scheme and query
-        $url = preg_replace('|(\?.*)?$|', '', (string) $this);
-        $url = preg_replace('|^(\w+://)|', '', $url);
+        //Remove scheme
+        $url = preg_replace('|^(\w+://)|', '', (string) $this);
```

Comment:
I know, but it's also essential to be able to block urls with specific parameters, witch is the main goal with this PR.
The obvious alternative is to re-invent `match()`, and specifying an blacklist for each part of the url (scheme, host, port, parameter, query).

The definite best solution would be switching over to unconditional regex, letting the regex string specify every requirement, but again, that may also have unintended consequences for the minority.

How about double configuration blacklists then? and deprecate the old one?
- `images_blacklist` and `image_blacklist_regex`
- `url_blacklist_regex`